File: backend/src/models/EmailAddress.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const EmailAddress = sequelize.define('EmailAddress', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
    allowNull: false,
  },
  organization_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  email: {
    type: DataTypes.STRING(255),
    allowNull: false,
  },
  normalized: {
    type: DataTypes.STRING(255),
    allowNull: true,
  },
  email_type: {
    type: DataTypes.STRING(20),
    defaultValue: 'GENERAL',
  },
  source_url: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
  is_valid: {
    type: DataTypes.BOOLEAN,
    defaultValue: true,
  },
  created_at: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW,
  },
}, {
  tableName: 'email_addresses',
  timestamps: false,
});

module.exports = EmailAddress;

File: backend/app/models/email_address.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime, ForeignKey, Text, func
from sqlalchemy.orm import relationship
from app.database.database import Base


class EmailAddress(Base):
    __tablename__ = "email_addresses"

    id = Column(Integer, primary_key=True, index=True)
    organization_id = Column(Integer, ForeignKey("organizations.id"), nullable=False)

    email = Column(String(255), nullable=False)
    normalized = Column(String(255), nullable=True)
    email_type = Column(String(20), default="GENERAL")  # INFO/CONTACT/ADMISSIONS/SUPPORT/GENERAL/UNKNOWN
    source_url = Column(Text, nullable=True)
    is_valid = Column(Boolean, default=True)
    created_at = Column(DateTime, server_default=func.now())

    organization = relationship("Organization", back_populates="email_addresses")
